tests/e2e/hcp_machine_pool_test.go

```diff
@@ -929,6 +964,44 @@ var _ = Describe("HCP MachinePool", ci.Day2, ci.FeatureMachinepool, func() {
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).To(MatchRegexp(`KubeletConfig with name 'notexisting'[\n\t\s]*does not exist for cluster`))
 		})
+
+		It("imdsv2 fields - [id:75393]", ci.Medium, func() {
+			By("Try to create a nodepool with invalid imdsv2")
+			mpName := helper.GenerateRandomName("np-75393", 2)
+			validateMPArgAgainstErrorSubstrings(mpName, func(args *exec.MachinePoolArgs) {
+				args.Ec2MetadataHttpTokens = helper.StringPointer("invalid")
+			}, "Expected a valid param. Options are [optional required]. Got invalid.")
+
+			reqBody := []map[string]string{
+				{
+					"create_value": "required",
+					"edit_value":   "optional",
+				},
+				{
+					"create_value": "optional",
+					"edit_value":   "required",
+				},
+			}
+
+			for _, imdsv2Value := range reqBody {
+				By("Create a machinepool with --ec2-metadata-http-tokens = " + imdsv2Value["create_value"])
+				mpName := helper.GenerateRandomName("np-75393", 2)
+				mpArgs := getDefaultMPArgs(mpName)
+				mpArgs.Ec2MetadataHttpTokens = helper.StringPointer(imdsv2Value["create_value"])
+				_, err := mpService.Apply(mpArgs)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Try to edit nodepool with ec2_metadata_http_tokens = " + imdsv2Value["edit_value"])
+				errMsg := fmt.Sprintf(
+					`Attribute aws_node_pool.ec2_metadata_http_tokens, cannot be changed from "%s" to "%s"`,
+					imdsv2Value["create_value"], imdsv2Value["edit_value"])
+				validateMPArgAgainstErrorSubstrings(mpName, func(args *exec.MachinePoolArgs) {
+					args.Ec2MetadataHttpTokens = helper.StringPointer(imdsv2Value["edit_value"])
+				}, errMsg)
+
+				mpService.Destroy()
```

Comment:
This should be in a defer, just after the MP creation

Reply:
It doesn't work there @radtriste